django_project/frontend/views/users.py

```diff
@@ -0,0 +1,310 @@
+                if not org_user.user == self.request.user:
+                    data.append({
+                        'organisation': str(org_user.organisation),
+                        'user': str(org_user.user),
+                        'id': org_user.user.id,
+                        'role': role
+                        # Add more columns as needed
+                    })
+
+        return JsonResponse({'data': json.dumps(data, cls=DjangoJSONEncoder)})
+
+    def post(self, request):
+        # Default post method logic
+        return JsonResponse({'status': 'success'})
+
+    def invite_post(self, request):
+
+        # retrieve data from front end/html template
+        email = request.POST.get('email')
+        role = request.POST.get('inviteAs')
+        permissions = request.POST.get('memberRole')
+
+        # assign role from the roles defined in the db
+        if role == 'manager':
+            if permissions == 'write':
+                role = 'Admin'
+            else:
+                role = 'Base user'
+        else:
+            if permissions == 'write':
+                role = 'Admin'
+            else:
+                role = 'Base user'
+
+        # get role by name
+        user_role = self.get_user_role(role)
+
+        try:
+            # add invitation to model
+            is_new_invitation = self.is_new_invitation(
+                email, self.request.session[CURRENT_ORGANISATION_ID_KEY])
+            org_id = self.request.session[CURRENT_ORGANISATION_ID_KEY]
+            if not is_new_invitation:
+                if role == 'manager':
+                    create_invite = OrganisationInvites(
+                        email=email,
+                        organisation_id=org_id,
+                        user_role=user_role,
+                        assigned_as=OrganisationInvites.MANAGER
+                    )
+                else:
+                    create_invite = OrganisationInvites(
+                        email=email,
+                        organisation_id=org_id,
+                        user_role=user_role,
+                        assigned_as=OrganisationInvites.MEMBER
+                    )
+
+
+                set_request(request)  # Set the request object
+                create_invite.save()
+                
+             
+                invites = self.get_organisation_invites()
+                print(invites)
+                serialized_invites = json.dumps(list(invites))
+                return JsonResponse(
+                    {
+                        'status': 'success',
+                        'updated_invites': serialized_invites
+                    }
+                )
+            else:
+                return JsonResponse({'status': 'invitation already sent'})
+        except Exception as e:
+            print(f"Error creating invite: {str(e)}")
+            return JsonResponse({'status': 'invitation already sent'})
+
+
+
+
+    def delete_post(self, request):
```

Comment:
Please improve this function, because currently if an error occurs (like the object not existing), the code returns a success status.